### PR TITLE
Error case for NULL arg causing segfault in iocshArgPersistentString

### PR DESCRIPTION
**Pull request by ghost**
As per [bug report](https://bugs.launchpad.net/epics-base/+bug/1824732), `iocshPersistentString` will check if the given arg is `NULL` before any C lib functions are used against it which can cause segfaults.

### CODE REVIEW

File: modules/libcom/src/iocsh/iocsh.cpp [hunk 1]
```diff
@@ -314,13 +314,18 @@ cvtArg (const char *filename, int lineno, char *arg, iocshArgBuf *argBuf,
...
+        if (arg != NULL) {
+            argBuf->sval = (char *) malloc(strlen(arg) + 1);
+            if (argBuf->sval == NULL) {
+                showError(filename, lineno, "Out of memory");
+                return 0;
+            }
+            strcpy(argBuf->sval, arg);
+            break;
+        } else {
+            showError(filename, lineno, "Unable to copy as provided arg was NULL");
```
Comment (mdavidsaver) [inline modules/libcom/src/iocsh/iocsh.cpp]: I don't think that `!arg` is an error and rather should be passed through as `!argBuf->sval`.

Comment (ghost) [reply modules/libcom/src/iocsh/iocsh.cpp]: I've updated this now so `argBuf->sval = NULL` in the event a null argument is passed.

Should we also test for `argBuff == NULL`? Or is it implicit that `argBuf` cannot be NULL?

Freddie also mentioned that we could potentially use the epics strdup?

**Comment by ghost** [comment]
I've added a test case, but I've passed the argument in the cmd as `epicsThreadSleep \0`, I have a feeling this will be interpreted as an empty string. What route would you recommend in order to test a NULL input for the `iocshArgPersistentString`?

File: modules/libcom/test/iocshTestNullArgument.cmd [hunk 2]
```diff
@@ -0,0 +1,4 @@
+on error break
+position before_error
+epicsThreadSleep \0
```
Comment (mdavidsaver) [inline modules/libcom/test/iocshTestNullArgument.cmd]: This line isn't doing what you think.  This is passing `"\\0"`.  Pass nothing to pass an NULL.  eg. just `epicsThreadSleep` with no number.  That said, this does doesn't cover the change to `iocsh.cpp` as argument type of epicsThreadSleep isn't iocshArgPersistentString.  (as mentioned on the LP bug, there are no internal users)

**Comment by mdavidsaver** [comment]
The change to `iocsh.cpp` now looks fine.

File: modules/libcom/test/iocshTestNullArgument.cmd (hunk 2 above)
Comment (ghost) [reply modules/libcom/test/iocshTestNullArgument.cmd]: Thanks. This clarifies my previous message.

**Comment by mdavidsaver** [comment]
Applied as 45bbe274e9f13f935a5e31ba2e7cc46e26495e7f and 6e0706a3121a56ba395db1fcb28ba58c5c623e81
